process/Style-guide.md

```diff
@@ -0,0 +1,250 @@
+
+* Use `return None` instead of `return` when your function or method is expected to return a value.
+* Do not use `return None` in generators.
+* Use `return` in a function or method that does not return a value.
+
+## Docstrings
+
+* Use English, and avoid abbreviations. Use "for example" or "such as" instead of Latin abbreviations like "e.g.".
+  * Acronyms and initialisms should be preserved, for example HTMLParser and not HtmlParser.
+* We use "Google Style" docstrings. 
+  * For examples, see [this page](http://sphinxcontrib-napoleon.readthedocs.org/en/latest/example_google.html) as well as the notes below. 
+
+There are still a few legacy docstrings in the codebase, here are some examples you might see. Please don't write new code that looks like this:
+
+```python
+def AddAnalysisReport(self, analysis_report):
+    """Adds an analysis report.
+    
+    Args:
+      analysis_report: a report.
+    """
+```
+This is missing an important detail, the argument type. Is it a string? Some other sort of object? How about this:
+
+```python
+def AddAnalysisReport(self, analysis_report):
+    """Adds an analysis report.
+    
+    Args:
+      analysis_report: an analysis report object (instance of AnalysisReport)
+    """
+```
+This is overly verbose, and is hard to parse.
+
+Instead do:
+
+```python
+def AddAnalysisReport(self, analysis_report, storage_writer=None):
+    """Adds an analysis report.
+    
+    Args:
+      analysis_report (AnalysisReport): a report.
+      storage_writer (Optional[StorageWriter]): the storage writer must be open, 
+          and must not be closed. If no storage_writer is provided, a new writer 
+          will be created.
+    """
+```
+
+Make sure your arguments descriptions include:
+
+1. The argument(s) type(s);
+2. In case of [standard types](https://docs.python.org/3.3/library/stdtypes.html) a description of their format. Note that we use the Python 3 standard types;
+3. Description of the meaning of the argument. In other words how the argument is used by the function (or method). If the description exceeds the line limit, indent the next line with 4 spaces.
+
+The meaning can be left out if the function has only a couple arguments and how the arguments are used is obvious from the description as in the example of `AddAnalysisReport`.
+
+A few other tips:
+
+### Compound types
+
+If a function deals with a compound type (list, dict), document it like so:
+```
+Args:
+  constraints (dict[str, Filter]): constraint name mapped to the filter that implements the constraint.
+
+Returns:
+  list[BaseParser]: all relevant parsers.
+```
+
+### Multiple acceptable types
+
+If you need to specify multiple types, use a pipe to separate them. For example:
+
+```
+Args:
+  path (str|Path): path to tag file.
+```
+
+### Multiple return types
+
+Python simulates multiple arguments being returned by implicitly returning a tuple. Document like so:
+
+```
+...
+Returns:
+  tuple: containing:
```

Comment:
Is this still accurate with the pylint docstring check?

Reply:
Yes.